libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts

```diff
@@ -1434,20 +1565,55 @@ const ParentalLeaveTemplate: ApplicationTemplate<
         unset(application.answers, 'previousState')
         return context
       }),
-      setPreviousState: assign((context) => {
+      setPreviousState: assign((context, event) => {
         const { application } = context
-        const { state, answers } = application
+        const { state } = application
+        const { answers } = application
+        const e = (event.type as unknown) as any
+        if (e === 'xstate.init') {
+          return context
+        }
```

Comment:
We has a problem here when we tested EditOrAddPeriods. Is it not problem anymore?
If not, just ignore my comment

Reply:
It seemed like everything was working fine. So I don't think there is a problem here.
But I will try it a couple of times more.